### PR TITLE
add cloudwatch metric alarm

### PR DESCRIPTION
**Pull request by pjaudiomv**
plan

```hcl
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # aws_cloudwatch_metric_alarm.vphone_healthy_hosts will be created
  + resource "aws_cloudwatch_metric_alarm" "vphone_healthy_hosts" {
      + actions_enabled                       = true
      + alarm_actions                         = (known after apply)
      + alarm_description                     = "Zero healthy nodes in Target Group"
      + alarm_name                            = "vphone-healthy-hosts"
      + arn                                   = (known after apply)
      + comparison_operator                   = "LessThanThreshold"
      + dimensions                            = {
          + "LoadBalancer" = "app/vphone/8504c6feac3801f3"
          + "TargetGroup"  = "targetgroup/vphone/654f9785cf3b9ab8"
        }
      + evaluate_low_sample_count_percentiles = (known after apply)
      + evaluation_periods                    = 1
      + id                                    = (known after apply)
      + metric_name                           = "HealthyHostCount"
      + namespace                             = "AWS/ApplicationELB"
      + ok_actions                            = (known after apply)
      + period                                = 60
      + statistic                             = "Sum"
      + threshold                             = 1
      + treat_missing_data                    = "missing"
    }

  # module.notify_slack.data.aws_iam_policy_document.lambda[0] will be read during apply
  # (config refers to values not yet known)
 <= data "aws_iam_policy_document" "lambda"  {
      + id   = (known after apply)
      + json = (known after apply)

      + statement {
          + actions   = [
              + "logs:CreateLogStream",
              + "logs:PutLogEvents",
            ]
          + effect    = "Allow"
          + resources = [
              + (known after apply),
            ]
          + sid       = "AllowWriteToCloudwatchLogs"
        }
    }

  # module.notify_slack.aws_cloudwatch_log_group.lambda[0] will be created
  + resource "aws_cloudwatch_log_group" "lambda" {
      + arn               = (known after apply)
      + id                = (known after apply)
      + name              = "/aws/lambda/notify_slack"
      + retention_in_days = 0
    }

  # module.notify_slack.aws_iam_role.lambda[0] will be created
  + resource "aws_iam_role" "lambda" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Effect    = "Allow"
                      + Principal = {
                          + Service = "lambda.amazonaws.com"
                        }
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + max_session_duration  = 3600
      + name                  = (known after apply)
      + name_prefix           = "lambda"
      + path                  = "/"
      + unique_id             = (known after apply)
    }

  # module.notify_slack.aws_iam_role_policy.lambda[0] will be created
  + resource "aws_iam_role_policy" "lambda" {
      + id          = (known after apply)
      + name        = (known after apply)
      + name_prefix = "lambda-policy-"
      + policy      = (known after apply)
      + role        = (known after apply)
    }

  # module.notify_slack.aws_lambda_function.notify_slack[0] will be created
  + resource "aws_lambda_function" "notify_slack" {
      + arn                            = (known after apply)
      + description                    = "Send Alerts to Slack"
      + filename                       = ".terraform/modules/notify_slack/functions/notify_slack.zip"
      + function_name                  = "notify_slack"
      + handler                        = "notify_slack.lambda_handler"
      + id                             = (known after apply)
      + invoke_arn                     = (known after apply)
      + last_modified                  = (known after apply)
      + memory_size                    = 128
      + publish                        = false
      + qualified_arn                  = (known after apply)
      + reserved_concurrent_executions = -1
      + role                           = (known after apply)
      + runtime                        = "python3.7"
      + source_code_hash               = "cDWWIGJenJLv7mMUx1rpbzNVWli0VHuWmqDHqnfuEAc="
      + source_code_size               = (known after apply)
      + timeout                        = 30
      + version                        = (known after apply)

      + environment {
          + variables = {
              + "LOG_EVENTS"        = "False"
              + "SLACK_CHANNEL"     = "virtual-phone"
              + "SLACK_EMOJI"       = ":aws:"
              + "SLACK_USERNAME"    = "alert-bot"
              + "SLACK_WEBHOOK_URL" = "REDACTED_FROM_PLAN"
            }
        }

      + tracing_config {
          + mode = (known after apply)
        }
    }

  # module.notify_slack.aws_lambda_permission.sns_notify_slack[0] will be created
  + resource "aws_lambda_permission" "sns_notify_slack" {
      + action        = "lambda:InvokeFunction"
      + function_name = "notify_slack"
      + id            = (known after apply)
      + principal     = "sns.amazonaws.com"
      + source_arn    = (known after apply)
      + statement_id  = "AllowExecutionFromSNS"
    }

  # module.notify_slack.aws_sns_topic.this[0] will be created
  + resource "aws_sns_topic" "this" {
      + arn    = (known after apply)
      + id     = (known after apply)
      + name   = "slack-topic-alert"
      + policy = (known after apply)
    }

  # module.notify_slack.aws_sns_topic_subscription.sns_notify_slack[0] will be created
  + resource "aws_sns_topic_subscription" "sns_notify_slack" {
      + arn                             = (known after apply)
      + confirmation_timeout_in_minutes = 1
      + endpoint                        = (known after apply)
      + endpoint_auto_confirms          = false
      + id                              = (known after apply)
      + protocol                        = "lambda"
      + raw_message_delivery            = false
      + topic_arn                       = (known after apply)
    }

Plan: 8 to add, 0 to change, 0 to destroy
```

### CODE REVIEW

File: terraform/cloudwatch.tf [hunk 1]
```diff
@@ -2,3 +2,31 @@ resource aws_cloudwatch_log_group vphone {
   name              = "vphone"
   retention_in_days = 14
 }
+
+module "notify_slack" {
+  source             = "git@github.com:terraform-aws-modules/terraform-aws-notify-slack.git"
```
Comment (jbraswell) [inline terraform/cloudwatch.tf]: It looks in git by default - see https://github.com/mvascna/virtual-phone-line/blob/master/terraform/vpc.tf#L2

File: terraform/cloudwatch.tf [hunk 2]
```diff
@@ -2,3 +2,31 @@ resource aws_cloudwatch_log_group vphone {
   name              = "vphone"
   retention_in_days = 14
 }
+
+module "notify_slack" {
```
Comment (jbraswell) [inline terraform/cloudwatch.tf]: Follow conventions for the rest of the project and remove double quotes from resource names.

**Review by jbraswell — DISMISSED** [review]

File: terraform/cloudwatch.tf (hunk 1 above)
Comment (pjaudiomv) [reply terraform/cloudwatch.tf]: nah dawg

Comment (pjaudiomv) [reply terraform/cloudwatch.tf]: it looks in terraform registry by default

**Pull request merged**